restli-int-test/src/test/java/com/linkedin/restli/examples/TestGreetingUnstructuredDataResources.java

```diff
@@ -114,4 +112,32 @@ public void testGetInternalServiceException(String resourceURL)
       assertEquals(conn.getResponseCode(), 500);
     });
   }
+
+  @DataProvider(name = "exceptionNoContentURLs")
+  private static Object[][] exceptionNoContentURLs()
+  {
+    return new Object[][] {
+        { "/greetingAssociationUnstructuredData/src=exception_204&dest=bar" }
+    };
+  }
+
+  @Test(dataProvider = "exceptionNoContentURLs")
+  public void testGetInternalServiceExceptionWithNoContent(String resourceURL)
+      throws Throwable
+  {
+    sendGet(resourceURL, (conn) -> {
+      conn.setDoInput(true);
+      int serverStatusCode = conn.getResponseCode();
+      assertEquals(serverStatusCode, 204);
+      BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
+      String inputLine;
+      StringBuffer response = new StringBuffer();
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      in.close();
+      assertEquals(response.length(), 0);
+      assertTrue(conn.getHeaderField("X-LinkedIn-Error-Response").equalsIgnoreCase("true"));
```

Comment:
I am kind of confused for this test case. 
204 is not an error code, but in the test resource you wrapped it in a RestLiServiceException. In the assertion, the X-LinkedIn-Error-Response = true, is this the right expectation?

Reply:
It is not, but it is the current behavior. This is why we added the error log, which can help migrate the existing usecases before we start failing for such cases.

Reply:
There might be some question about why this test exists. Along with the other 2 tests this seems redundant.
But actually, the http client in this test is pure, not decorated with HttpBridges and callbacks.
These bridges and callbacks might change an returned error to an RemoteInvokeException, And I am afraid I am still not fully grab the whole picture.

Reply:
One value this adds is that it checks we still send the error response header when server threw RestLiServiceException with 204 status, as that is the current behavior.

Reply:
Thanks for explaining it. @karthikbalasub